### PR TITLE
Fix trust graph

### PR DESCRIPTION
**Pull request by xoriole**
Fixes https://github.com/Tribler/tribler/issues/5675

This PR updates the Trust graph previously based on Trustchain blocks to use new bandwidth accounting transaction blocks.

Major changes
- The graph only compose two layers, layer 1 (friends), layer 2 (friend of friends)
- GUI and REST API removes the depth parameter
- Auto refreshing on the GUI is removed
- Test fixes

trust-graph-new

### CODE REVIEW

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

File: src/tribler-core/tribler_core/modules/bandwidth_accounting/database.py [hunk 1]
```diff
@@ -87,6 +87,16 @@ def get_latest_transaction(self, public_key_a: bytes, public_key_b: bytes) -> Ba
...
         return BandwidthTransactionData.from_db(db_obj) if db_obj else None
 
+    @db_session
+    def get_latest_transactions(self, public_key: bytes) -> List[BandwidthTransactionData]:
+        """
+        Return the latest transactions of a given public key, or an empty list if no transactions exist.
+        :param public_key: The public key of the party transferring the bandwidth.
+        :return The latest transactions of the specified public key, or an empty list if no transactions exist.
+        """
+        db_objs = self.BandwidthTransaction.select(lambda bt: public_key in (bt.public_key_a, bt.public_key_b))[:]
```
Comment (drew2a) [inline src/tribler-core/tribler_core/modules/bandwidth_accounting/database.py:97]: Is the `[:]` necessary here?

Seems like `db_objs` will be used only for iterating.

Comment (drew2a) [reply src/tribler-core/tribler_core/modules/bandwidth_accounting/database.py:97]: Did you check this request on large databases? Transaction's DB can be large...

File: src/tribler-core/tribler_core/modules/trust_calculation/trust_graph.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        max_x = max_y = 0.0001  # as close to zero
+        for _id, (theta, r) in pos.items():
+            index_mapper[_id] = node_id
+            node = undirected_graph.nodes()[_id]
+            node['id'] = node_id
+            node_id += 1
+
+            # convert from polar coordinates to cartesian coordinates
+            x = r * math.sin(theta) * num_nodes
+            y = r * math.cos(theta) * num_nodes
```
Comment (drew2a) [inline src/tribler-core/tribler_core/modules/trust_calculation/trust_graph.py:125]: wow! you do it manually...

**Review by drew2a — DISMISSED** [review]
I suggest you check how it works on large data.

File: src/tribler-core/tribler_core/modules/bandwidth_accounting/database.py (hunk 1 above)
Comment (xoriole) [reply src/tribler-core/tribler_core/modules/bandwidth_accounting/database.py:97]: - Good point. `[:]` is indeed not necessary.
- The number of transactions is small compared to previous Trustchain database unless there are thousands of nodes but a good point to add a limit. I'll make the change.

File: src/tribler-core/tribler_core/modules/trust_calculation/trust_graph.py (hunk 2 above)
Comment (xoriole) [reply src/tribler-core/tribler_core/modules/trust_calculation/trust_graph.py:125]: Credit goes to @cuileri for the first implementation.

**Comment by xoriole** [comment]
The three security hotspots reported by sonarcloud are not security issues in context of this PR.
-  `sha3_224()` I consider sha3 safe to use. I'm not sure why sonarcloud is complaining.
- `random.randint(1, 100)` this is used in test and not a security threat

**Review by qstokkink — APPROVED** [review]

**Pull request merged**